tests/testthat/test-strategy-ff16w.R

```diff
@@ -52,31 +56,43 @@ test_that("FF16w Environment", {
   
 })
 
-test_that("Basic run", {
+test_that("Rainfall spline basic run", {
   # one species
   p0 <- scm_base_parameters("FF16w")
   p1 <- expand_parameters(trait_matrix(0.0825, "lma"), p0, FF16w_hyperpar,FALSE)
   p1$birth_rate <- 20
-
+  
   env <- make_environment("FF16w", 
                           soil_number_of_depths = 10,
-                          soil_initial_state = rep(1, 10),
-                          soil_infiltration_rate = 1)
+                          soil_initial_state = rep(1, 10))
+  
+  # init rainfall spline for env
+  x <- seq(0, 110, 1)
+  integrand <- function(x) {x^2}
+  y <- integrand(x)
+  env$rainfall_init(x, y)
   
   ctrl <- scm_base_control()
   
   out <- run_scm(p1, env, ctrl)
   
   expect_equal(out$patch$environment$ode_size, 10)
-
+  
+  # model is currently 1/z * int(0, t)(f(x)), where z is depth, 
+  # x is time, f is spline, t is number of years
+  
+  # check the rates are correct, ie 105^2/depth
   expect_equal(out$patch$environment$soil$rates, 
-               c(1.00, 0.50, 0.33, 0.25, 0.20, 0.17, 0.14, 0.13, 0.11, 0.10),
+               sapply(seq(1, 10), function(x) { (out$time^2)/x }),
                tolerance = .01)
   
+  # check the states are correct
+  # model is currently 1/z * int(0, t)(f(x)), where z is depth, 
+  # x is time, f is spline, t is number of years
   expect_equal(out$patch$environment$soil$states,
-               c(105, 52, 35, 26, 21, 17, 15, 13, 11, 10),
+               sapply(seq(1, 10), function(i) { (1/i) * integrate(integrand, 0, out$time)$value }),
                tolerance = 0.1)
```

Comment:
again with tolerance

Reply:
Was able to pass this with tolerance `1e-5` by using step value of 0.1:
```
x <- seq(0, 110, 0.1)
```
This is obviously going to slow the tests down a little bit since the spline will be created with 1100 control points, but the numbers for this test are quite large so accuracy would unsurprisingly suffer. Trying to use tolerance `1e-6` yields:
```
10/10 mismatches (average diff: 1)
[1] 389415 - 389414 == 1
[2] 194708 - 194707 == 1
[3] 129806 - 129805 == 1
[4]  97354 -  97353 == 1
[5]  77884 -  77883 == 1
[6]  64903 -  64902 == 1
[7]  55632 -  55631 == 1
[8]  48678 -  48677 == 1
[9]  43269 -  43268 == 1
```
Maybe I could try with a lower order polynomial than quadratic...